components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -0,0 +1,742 @@
+
+            // Validate the requested_expiry of the Request.
+            validateRequestedExpiry(claimsSet);
+
+            if (log.isDebugEnabled()) {
+                log.debug(" CIBA Authentication Request made by client with clientID : " +
+                        claimsSet.getIssuer() + "is properly validated.");
+            }
+
+        } catch (ParseException e) {
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.SERVER_ERROR,
+                    "error in validating authentication request.",
+                    e);
+        }
+    }
+
+    /**
+     * Checks whether the requested_expiry values exists and is valid.
+     *
+     * @param claimsSet JWT claimsets of the authentication request.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Server Exception.
+     */
+    private void validateRequestedExpiry(JWTClaimsSet claimsSet) throws CibaAuthFailureException {
+
+        // Validation for requested_expiry.
+        if ((claimsSet.getClaim(CibaConstants.REQUESTED_EXPIRY)) == null) {
+            //  Requested_expiry claim value does not exist.
+            return;
+        }
+        if (StringUtils.isBlank(String.valueOf(claimsSet.getClaim(CibaConstants.REQUESTED_EXPIRY)))) {
+            // Requested expiry is a blank value.
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                        claimsSet.getIssuer() + " .The request is with invalid  value for " +
+                        "(requested_expiry).");
+            }
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST,
+                    "invalid value for (requested_expiry).");
+        }
+    }
+
+    /**
+     * Checks whether the transaction_context values exists and is valid.
+     *
+     * @param claimsSet JWT claimsets of the authentication request.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Server Exception.
+     */
+    private void valiateTransactionContext(JWTClaimsSet claimsSet) throws CibaAuthFailureException {
+
+        try {
+            // Validation for transaction_context.
+            if ((claimsSet.getClaim(CibaConstants.TRANSACTION_CONTEXT)) == null) {
+                // Request has no transaction_context claim.
+                return;
+            }
+            if (StringUtils.isBlank(claimsSet.getJSONObjectClaim(CibaConstants.TRANSACTION_CONTEXT).toJSONString())) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Invalid CIBA Authentication Request made by client with clientID : " +
+                            claimsSet.getIssuer() + ".The request is with invalid  " +
+                            "value for (transaction_context).");
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST,
+                        "invalid value for (transaction_context).");
```

Comment:
Let's maintain a consistent error message format. some error message starts with Capital letter and some not.